test/integration/scheduler_perf/util.go

```diff
@@ -117,34 +116,34 @@ func mustSetupCluster(ctx context.Context, tb testing.TB, config *config.KubeSch
 		var err error
 		config, err = newDefaultComponentConfig()
 		if err != nil {
-			tb.Fatalf("Error creating default component config: %v", err)
+			tCtx.Fatalf("Error creating default component config: %v", err)
 		}
 	}
 
-	client := clientset.NewForConfigOrDie(cfg)
-	dynClient := dynamic.NewForConfigOrDie(cfg)
+	kCtx := tCtx.KContext()
+	ktesting.SetClients(kCtx, cfg)
 
 	// Not all config options will be effective but only those mostly related with scheduler performance will
 	// be applied to start a scheduler, most of them are defined in `scheduler.schedulerOptions`.
-	_, informerFactory := util.StartScheduler(ctx, client, cfg, config)
-	util.StartFakePVController(ctx, client, informerFactory)
-	runGC := util.CreateGCController(ctx, tb, *cfg, informerFactory)
-	runNS := util.CreateNamespaceController(ctx, tb, *cfg, informerFactory)
+	_, informerFactory := util.StartScheduler(kCtx, kCtx, cfg, config)
+	util.StartFakePVController(kCtx, kCtx, informerFactory)
+	runGC := util.CreateGCController(kCtx, kCtx, *cfg, informerFactory)
+	runNS := util.CreateNamespaceController(kCtx, kCtx, *cfg, informerFactory)
```

Comment:
This section shows that a lot of code in `k8s.io/kubernetes/test/integration/util` would benefit from conversion to the new context types - that will get rid of the redundant parameters.

Reply:
In `TContext`, the client interface is no longer part of the interface itself, so this code now reads like this:
```
runNS := util.CreateNamespaceController(kCtx, kCtx.Client(), *cfg, informerFactory)
```

I find that a bit less confusing when it occurs, although removing such redundant parameters should still be the ultimate goal.